### ISSUE TITLE
[feat]: integrate WebGazer - uptime monitoring and analytics

### ISSUE THREAD

**Issue by th0th** [gatsbyjs/gatsby#12653]
I am not sure if this is the right place to post this but I couldn't find a better way to contact, so, here it is...

I am an indie-ish hacker and built the [WebGazer - Uptime monitoring and analytics platform](https://www.webgazer.io). And I want to support the free/open source projects that enabled me to build WebGazer. It was such a relief to migrate previous jQuery'ied website to React with Gatsby, I still remember. Even first stable version wasn't out, I've been using gatsby since then. I'd love to financially contribute, too, but WebGazer isn't making much yet :)

tl;dr

I want to provide a free subscription, if it helps, with all features enabled.

I am not sure about the current stack of Gatsby, saw Netlify on the README but I think even if not public web pages, API endpoints used by Gatsby Store could benefit from HTTP monitoring. Other than that, if there are any scheduled tasks like cron jobs and such, Pulse would be useful.

**Comment by DSchau** [comment]
Sure - we can definitely check this out.

How does it work? Performance is a priority (™️) for Gatsby, so if we can avoid any heavy client-side scripts, would certainly prefer to go that route.

What would it take to begin trialing this and setting it up?

**Comment by th0th** [comment]
Thank you @DSchau, glad to hear :)

I will upgrade the account once you sign up on the website. Then for the HTTP monitoring, we will need to add URLs to be monitored, these monitors are called **gazers**. We add a gazer, set the URL and request parameters (request method, headers, data etc.) if necessary and we are done. No client-side alterations needed! Then WebGazer sends a request every minute and checks if the response is OK. If not, it alerts. E-mail, Slack, webhook, SMS and phone call alerts are available, not sure which one would suit though. With no alerts set, we can just review the statistics, up to you :)

About the performance monitoring part, we can set a response time threshold, let's say we want to know if there were 3 requests that have response time >= 1 second in the last 30 mins. Then again, we get an underperforming alert if it happens. I think we can enable performance monitoring later, let's see the average response times first.

Aaand we can also set a status page similar to [this one](https://status.webgazer.io). Web application part requires authentication, but status pages are publicly accessible.

Whoa, it sounded like a lengthy pitch. Let's start, you can sign up on the website. I will help with the configurations.
